File: testautomation/Scenario.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Chrome;

namespace SeleniumTests
{
    public class Scenario
    {

        protected IWebDriver driver;
        protected StringBuilder verificationErrors;
        protected string baseURL;
        protected bool acceptNextAlert = true;

        [SetUp]
        public void SetupTest()
        {
            driver = new ChromeDriver(@"C:\Users\Administrator\.nuget\packages\Selenium.WebDriver.ChromeDriver\2.24.0\driver");
            baseURL = "https://automatyzacja2016.wordpress.com/wp-admin";
            driver.Manage().Window.Maximize();
            verificationErrors = new StringBuilder();
        }

        [TearDown]
        public void TeardownTest()
        {
            try
            {
                driver.Quit();
            }
            catch (Exception)
            {
                // Ignore errors if unable to close the browser
            }
            Assert.AreEqual("", verificationErrors.ToString());
        }

        public void LogIn()
        {
            driver.Navigate().GoToUrl(baseURL);

            Insert(By.Id("user_login"), "szkolenieautomatyzacja");

            Insert(By.Id("user_pass"), "QW12qw12");

            Click(By.Id("wp-submit"));

        }
        public void LogOut()
        {
            Click(By.XPath("//*[@id=\"wp-admin-bar-my-account\"]/a/img"));
            Click(By.XPath("//*[@id=\"wp-admin-bar-user-info\"]/div/form/button"));
        }


        protected void Click(By by)
        {
            WebDriverWait waitThree = new WebDriverWait(driver, new TimeSpan(
                0, 0, 3));
            waitThree.Until(ExpectedConditions.ElementToBeClickable(by));
            driver.FindElement(by).Click();
        }

        protected void Insert(By by, string v)
        {
            driver.FindElement(by).Clear();
            driver.FindElement(by).SendKeys(v);
        }


        protected bool IsElementPresent(By by)
        {
            try
            {
                driver.FindElement(by);
                return true;
            }
            catch (NoSuchElementException)
            {
                return false;
            }
        }

        protected bool IsAlertPresent()
        {
            try
            {
                driver.SwitchTo().Alert();
                return true;
            }
            catch (NoAlertPresentException)
            {
                return false;
            }
        }

        protected string CloseAlertAndGetItsText()
        {
            try
            {
                IAlert alert = driver.SwitchTo().Alert();
                string alertText = alert.Text;
                if (acceptNextAlert)
                {
                    alert.Accept();
                }
                else
                {
                    alert.Dismiss();
                }
                return alertText;
            }
            finally
            {
                acceptNextAlert = true;
            }
        }
    }
}
        
    

File: src/test/java/Scenario.java
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import static org.junit.Assert.fail;

/**
 * Created by Administrator on 2016-09-22.
 */
public class Scenario {
    protected WebDriver driver;
    protected String baseUrl;
    protected boolean acceptNextAlert = true;
    protected StringBuffer verificationErrors = new StringBuffer();

    @Before
    public void setUp() throws Exception {


        System.setProperty("webdriver.chrome.driver", "C:\\chromedriver.exe");

        driver = new ChromeDriver(); //("C:\\chromedriver.exe");
        baseUrl = "https://automatyzacja2016.wordpress.com/";
        driver.get(baseUrl + "wp-admin");
        driver.manage().window().maximize();

    }
    protected void login() {
        insert(By.id("user_login"), "szkolenieautomatyzacja");
        insert(By.id("user_pass"), "QW12qw12");
        click(By.id("wp-submit"));
    }
    protected void assertAddPost() {
        click(By.xpath(" .//*[@id='primary']/div/div[2]/div[2]/div[2]/div[1]/div[3]/div/button[1]"));
        // Thread.sleep(1000);
        click(By.xpath(".//*[@id='primary']/div/div[2]/div[1]/div[2]/div[2]/div/div/a"));
        // Thread.sleep(3000);

        Assert.assertTrue("jakubg", driver.getPageSource().contains("jakubg"));
        System.out.println("Jest git!!!!!!!!!!!!!!!!!");
    }

    protected void addNewPost() {
        click(By.xpath(".//*[@id='wp-admin-bar-ab-new-post']/a"));
        //  Thread.sleep(3000);
        xpathSendKey(".//*[@id='primary']/div/div[2]/div[1]/div[2]/div[3]/div/textarea");
    }

    protected void xpathSendKey(String xpathExpression) {
        WebDriverWait waitN = new WebDriverWait(driver, 6);
        waitN.until(ExpectedConditions.elementToBeClickable(By.xpath(xpathExpression)));
        driver.findElement(By.xpath(xpathExpression)).sendKeys("jakubg");
    }

    protected void click(By xpath) {
        WebDriverWait wait = new WebDriverWait(driver, 3);
        wait.until(ExpectedConditions.elementToBeClickable(xpath));
        driver.findElement(xpath).click();
    }

    protected void insert(By login, String text) {
        driver.findElement(login).clear();
        driver.findElement(login).sendKeys(text);
    }


    //than


    @After
    public void tearDown() throws Exception {
        driver.quit();
        String verificationErrorString = verificationErrors.toString();
        if (!"".equals(verificationErrorString)) {
            fail(verificationErrorString);
        }
    }

    protected boolean isElementPresent(By by) {
        try {
            driver.findElement(by);
            return true;
        } catch (NoSuchElementException e) {
            return false;
        }
    }

    protected boolean isAlertPresent() {
        try {
            driver.switchTo().alert();
            return true;
        } catch (NoAlertPresentException e) {
            return false;
        }
    }

    protected String closeAlertAndGetItsText() {
        try {
            Alert alert = driver.switchTo().alert();
            String alertText = alert.getText();
            if (acceptNextAlert) {
                alert.accept();
            } else {
                alert.dismiss();
            }
            return alertText;
        } finally {
            acceptNextAlert = true;
        }
    }
}



